### PR TITLE
Grid added

### PR DESCRIPTION
**Pull request by SimonSchmid**
allows to create a grid of patches of an image

### CODE REVIEW

File: src/main/java/net/imglib2/algorithm/grid/Patch.java [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+		if (!Intervals.contains(srcImage, raiPos)) {
+			throw new IndexOutOfBoundsException("IndexOutOfBoundsException");
+		}
+		RandomAccessibleInterval<T> raiPatch = Views.interval(srcImage, min, max);
+		// extend the source image if the patch and borders overlap
+		if (!Intervals.contains(srcImage, raiPatch)) {
+			raiPatch = Views.interval(Views.extendMirrorSingle(srcImage), min, max);
+		}
+
+		this.patch = Views.zeroMin(raiPatch);
```
Comment (axtimwalde) [inline src/main/java/net/imglib2/algorithm/grid/Patch.java]: A new `View` is generated whenever `Patch` is returned by `GridRandomAccess.get()`.  This makes the `Patch` class not only obsolete but redundant.  Try to implement a solution that returns `Views.offsetInterval(long[], long[]);` directly in `GridRandomAccess.get()`.

File: src/main/java/net/imglib2/algorithm/grid/GridRandomAccess.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+import net.imglib2.RandomAccess;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.Sampler;
+
+/**
+ * {@link RandomAccess} for a grid.
+ * 
+ * @author Simon Schmid (University of Konstanz)
+ */
+public class GridRandomAccess<T> implements RandomAccess<RandomAccessibleInterval<T>> {
```
Comment (axtimwalde) [inline src/main/java/net/imglib2/algorithm/grid/GridRandomAccess.java]: Extend net.imglib2.Point to remove all `Localizable` and `Positionable` cruft.

**Comment by axtimwalde** [comment]
@SimonSchmid thanks for the initiative.  Could you please check if what you are trying here isn't already possible with a combination of [`RectangleShape`](https://github.com/imglib/imglib2-algorithm/blob/master/src/main/java/net/imglib2/algorithm/neighborhood/RectangleShape.java#L73) or [`CenteredRectangleShape`](https://github.com/imglib/imglib2-algorithm/blob/master/src/main/java/net/imglib2/algorithm/neighborhood/CenteredRectangleShape.java#L89) in combination with [`SubsampleIntervalView`](https://github.com/imglib/imglib2/blob/master/src/main/java/net/imglib2/view/Views.java#L907) and some coordinate shifting with [`Views.offsetInterval(...)`](https://github.com/imglib/imglib2/blob/master/src/main/java/net/imglib2/view/Views.java#L676)?

File: src/main/java/net/imglib2/algorithm/grid/Patch.java (hunk 1 above)
Comment (SimonSchmid) [inline src/main/java/net/imglib2/algorithm/grid/Patch.java]: I didn't know about this method, thanks!

File: src/main/java/net/imglib2/algorithm/grid/GridRandomAccess.java (hunk 2 above)
Comment (SimonSchmid) [inline src/main/java/net/imglib2/algorithm/grid/GridRandomAccess.java]: Nice hint!

**Comment by SimonSchmid** [comment]
@axtimwalde Thanks for the comments. I'll try your proposed solution, that could work.

**Comment by dietzc** [comment]
Keep in mind that `RectangleShape` neighborhood are not `RandomAccessible`!

**Comment by axtimwalde** [comment]
@dietzc true, but shouldn't it be `RandomAccessible`?

**Comment by dietzc** [comment]
@axtimwalde yes, even if I don't know if `Shape`s in general should be or just special cases? Because, for cases where the `Shape` is not a complete `Interval` (so everything besides `RectangleShape`) you would have to have e.g. a `RandomAccessible<Pair<BoolType,T>>`. The `BoolType` can be used for inside-outside testing. But I haven't really thought this through. Maybe @tpietzsch can provide more insight?

**Comment by axtimwalde** [comment]
For now I suggest special case, independent of the Shape interface because Rectangles are intervals.  @dietzc & @SimonSchmid, are you fine with me closing this pull request?  This is too far from ready and should be discussed in an issue or elsewhere.

**Comment by dietzc** [comment]
You can close it. We will keep the code on our side until it's clear if `RectangleShape`s can be `RandomAccessible` or not. I will discuss this with @tpietzsch at the upcoming hackathon.

**Pull request closed without merging**